cluster/sync.go

```diff
@@ -3,34 +3,36 @@ package cluster
 import (
 	"strings"
 
+	"github.com/weaveworks/flux"
 	"github.com/weaveworks/flux/resource"
 )
 
 // Definitions for use in synchronising a cluster with a git repo.
 
-// SyncAction represents either the deletion or application (create or
-// update) of a resource.
-type SyncAction struct {
-	Delete resource.Resource // ) one of these
-	Apply  resource.Resource // )
-}
-
-type SyncDef struct {
-	// The actions to undertake
-	Actions []SyncAction
+// SyncSet groups the set of resources to be updated. Usually this is
+// the set of resources found in a git repo; in any case, it must
+// represent the complete set of resources, as garbage collection will
+// assume missing resources should be deleted. The name is used to
+// distinguish the resources from a set from other resources -- e.g.,
+// cluster resources not marked as belonging to a set will not be
+// deleted by garbage collection.
+type SyncSet struct {
```

Comment:
After reading the code more carefully I don't see the reason for `SyncSet` at all, particularly since we enforce the whole repository resources to be included for garbage collection and `Name` always being set to `git`

Reply:
Since this ends up being reflected in clusters, I would like to leave at least a bit of room for extending it to more than one set of resources. (If it were entirely internal, I wouldn't care).

Reply:
I see

Reply:
After thinking about it a bit more, as it is today you don't even have the possibility of splitting the resources in multiple sets (causing garbage collection to remove resources in the other sets) so I still think we should remove it (since it leads to think you can actually use multiple sets).

Reply:
The name is now calculated based on the git repo config, and supplied im the SyncSet rather than being constant. (In other words, it's necessary to label the SyncSet). Providing more than one, at different times, would be OK since the garbage collection only examines things with the name given -- so it's not misleading.